src/Service/ServiceExe.cs

```diff
@@ -1785,77 +1809,136 @@ private void RunTaskRunInContainer(ServerTaskRun hostRun)
+            }
+
+            throw new FactoryOrchestratorContainerException(Resources.NoContainerIdFound);
         }
 
-        public string GetContainerId()
+        public void UpdateContainerId()
         {
             try
-            {
-                // TODO: replace with an API call
-                var cmdiag = RunProcessViaCmd("cmdiag.exe", "list", 5000);
-                var containerGuidString = cmdiag.TaskOutput.Where(x => x.Contains("cmscontainerstaterunning", StringComparison.InvariantCultureIgnoreCase)).DefaultIfEmpty(null).FirstOrDefault();
-
-                if (containerGuidString != null)
-                {
-                    return containerGuidString.Split(',', StringSplitOptions.RemoveEmptyEntries).First().Trim();
+            {
+                // FactoryOS puts the container guid in SYSTEM\CurrentControlSet\Control\FactoryUserManager
+                using var reg = Registry.LocalMachine.OpenSubKey(_volatileFactoryOSContainerRegKey, false);
+
+                var guidStr = (string)reg.GetValue(_factoryOSContainerGuidValue, String.Empty);
+                if (!string.IsNullOrEmpty(guidStr))
+                {
+                    ContainerGuid = Guid.Parse(guidStr);
+                }
+                else
+                {
+                    ContainerGuid = Guid.Empty;
+                    _containerClient = null;
+                    throw new FactoryOrchestratorContainerException(Resources.NoContainerIdFound);
                 }
             }
-            catch (Exception e)
+            catch (Exception)
             {
-                throw new FactoryOrchestratorContainerException(Resources.NoContainerIdFound, null, e);
-            }
-
-            throw new FactoryOrchestratorContainerException(Resources.NoContainerIdFound);
+                ContainerGuid = Guid.Empty;
+                _containerClient = null;
+                throw new FactoryOrchestratorContainerException(Resources.NoContainerIdFound);
+            }
         }
 
-        public List<IPAddress> GetContainerIpAddresses()
+        public void UpdateContainerIpAddress()
         {
-            List<IPAddress> ips = new List<IPAddress>();
             try
-            {
-                var containerGuid = GetContainerId();
-                var cmdiag = RunProcessViaCmd("cmdiag.exe", $"exec {containerGuid} -runas administrator -command \"ipconfig.exe\"", 5000);
-                var ipStrings = cmdiag.TaskOutput.Where(x => x.Contains("IPv4 address", StringComparison.InvariantCultureIgnoreCase));
-
-                foreach (var ipString in ipStrings)
-                {
-                    if (ipString != null)
-                    {
-                        ips.Add(IPAddress.Parse(ipString.Split(':', StringSplitOptions.RemoveEmptyEntries).Last().Trim()));
-                    }
+            {
+                // FactoryOS puts the container IP in SYSTEM\CurrentControlSet\Control\FactoryUserManager

```

Comment:
hmm; i don't love this. I understand if its needed for now but we should file an item to track this (likely as a part of figuring out plans around container management)

Reply:
The only options I know of currently are this or pinvoking a private API. I'm not sure if I can get this via HNS, if I can, that is JSON based and C# friendly, but I dont want to spend cycles on that now.

I'll make an issue

Reply:
#23